### PR TITLE
Split out transactions server actions into separate files

### PR DESCRIPTION
**Pull request by nswai983**
Moving transactions server actions into separate files per issue [#1113](https://github.com/actualbudget/actual/issues/1113).

No functional changes in this PR.

FYI this is my first contribution to actual, so please let me know if I miss anything!

### CODE REVIEW

File: packages/loot-core/src/server/transactions/app.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import { TransactionsHandler } from './types/handlers';
+
+// Expose functions to the client
+export const app = createApp<TransactionsHandler>();
+
+app.method(
+  'transactions-batch-update',
+  mutator(async function ({ added, deleted, updated, learnCategories }) {
+    return withUndo(async () => {
+      const result = await batchUpdateTransactions({
```
Comment (ACWalker) [inline packages/loot-core/src/server/transactions/app.ts:21]: Think you can just return this directly

File: packages/loot-core/src/server/transactions/app.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import { TransactionsHandler } from './types/handlers';
+
+// Expose functions to the client
+export const app = createApp<TransactionsHandler>();
+
+app.method(
+  'transactions-batch-update',
+  mutator(async function ({ added, deleted, updated, learnCategories }) {
+    return withUndo(async () => {
```
Comment (ACWalker) [inline packages/loot-core/src/server/transactions/app.ts:20]:
```suggestion
    return withUndo(async () => {
      result = await batchUpdateTransactions({
        added,
        updated,
        deleted,
        learnCategories,
      });
    });
```

File: packages/loot-core/src/server/transactions/app.ts (hunk 1 above)
Comment (ACWalker) [inline packages/loot-core/src/server/transactions/app.ts:21]: Can just return directly here.

**Review by ACWalker — APPROVED** [review]

**Comment by nswai983** [comment]
@ACWalker there were some merge conflicts I addressed that kept the PR from merging. I addressed those and your prior comment about returning directly in the 'transactions-batch-update' handler.

**Comment by nswai983** [comment]
Ran into some issues with Partials as a result of some changes to main. I have addressed these issues, and the current commit is up to date and passes all checks.

**Review by matt-fidd — APPROVED** [review]
Thanks for this!

File: packages/loot-core/src/server/transactions/types/handlers.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// @ts-strict-ignore
```
[the remark this answers is not in the document]
Comment (nswai983) [reply packages/loot-core/src/server/transactions/types/handlers.ts:1]: Ah I hadn't checked the types here. I ran into issues with the types being 'any' in lint checks so I used the types I did based on context surrounding the code being used.

I can update the types to match the packages/loot-core/src/types/server-handlers.d.ts file.

[the remark this answers is not in the document]
Comment (nswai983) [reply packages/loot-core/src/server/transactions/types/handlers.ts:1]: Question: Should the non-typed params (transaction, options, etc.) remain untyped? Or should types be added to these? I used TransactionEntity for the transaction paramaters to add some clarity surrounding them.

[the remark this answers is not in the document]
Comment (jfdoming) [reply packages/loot-core/src/server/transactions/types/handlers.ts:1]: Makes sense. I think adding extra types is fine! As long as we can be sure they're the right types. Feel free to also leave off improving the types to a separate PR, these things can be pretty in depth

**Comment by matt-fidd** [comment]
Hey @nswai983 - how are you doing on marrying up the types between this and `packages/loot-core/src/types/server-handlers.d.ts`?

**Pull request closed without merging**